packages/clay-management-toolbar/src/ActiveState.tsx

```diff
@@ -0,0 +1,229 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import ClayButton from '@clayui/button';
+import ClayIcon from '@clayui/icon';
+import Item from './Item';
+import ItemList from './ItemList';
+import React from 'react';
+import {ClayCheckbox} from '@clayui/form';
+import {ClayDropDownWithBasicItems} from '@clayui/drop-down';
+import {IItem} from '@clayui/drop-down';
+import {IPropsBase, TClickEvent} from './types';
+import {LinkOrButton, sub} from '@clayui/shared';
+
+interface IActionItems extends IItem {
+	quickAction?: boolean;
+}
+
+interface IProps extends IPropsBase {
+	actionItems?: Array<IActionItems>;
+	clearSelectionURL?: string;
+	onClearSelectionClick?: TClickEvent;
+	onQuickActionClick?: (
+		e: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+		item: IActionItems
+	) => void;
+	onSelectAllClick?: TClickEvent;
+	selectAllURL?: string;
+	selectedItems?: number;
+	totalItems?: number;
+}
+
+const ActiveState: React.FunctionComponent<IProps> = ({
+	actionItems = [],
+	checkboxChecked = false,
+	clearSelectionURL,
+	disabled = false,
+	filterItems,
+	labels = {
+		allSelected: 'All Selected',
+		checkbox: 'Select Items',
+		clearSelection: 'Clear',
+		selectAll: 'Select all',
+		selected: 'Selected',
+		selectedItemsOfTotalItems: '{0} of {1}',
+	},
+	onCheckboxChange,
+	onClearSelectionClick,
+	onInfoButtonClick,
+	onQuickActionClick,
+	onSearchValueChange,
+	onSelectAllClick,
+	selectAllURL,
+	selectedItems = 0,
+	sortingOrder,
+	spritemap,
+	titlesLang = {
+		infoButton: 'Info',
+	},
+	totalItems = 0,
+}) => {
+	return (
+		<>
+			<ItemList expand>
+				{onCheckboxChange && (
+					<Item>
+						<ClayCheckbox
+							checked={checkboxChecked}
+							disabled={disabled}
+							label={
+								!filterItems &&
+								!sortingOrder &&
+								!onSearchValueChange
+									? labels.checkbox
+									: undefined
+							}
+							onChange={onCheckboxChange}
+						/>
+					</Item>
+				)}
+				<Item>
+					<span className="navbar-text">
+						{selectedItems === totalItems ? (
+							labels.allSelected
+						) : (
+							<>
+								{sub(labels.selectedItemsOfTotalItems, [
+									selectedItems,
+									totalItems,
+								])}{' '}
```

Comment:
Is this empty space necessary or should this spacing be applied via css?

Reply:
It is necessary, not to merge the sub result with `labels.selected`, I think it would be strange to add a space via CSS should not be a problem with css. Maybe we can add backtick and add a space at the end but it's not very explicit.

```javascript
`${sub(labels.selectedItemsOfTotalItems, [selectedItems, totalItems])} `;
```

Reply:
No worries, the explicit `{' '}` is fine. I just wanted to make sure.

Reply:
👍